xcode/Subconscious/Shared/Components/MainToolbar.swift

```diff
@@ -9,11 +9,86 @@ import Foundation
 import SwiftUI
 import ObservableStore
 
+struct SyncStatusView: View {
+    var status: ResourceStatus
+    
+    var color: Color {
+        switch status {
+        case .failed:
+            return Color.pending
+        case .succeeded:
+            return Color.success
+        case .pending:
+            return Color.pending
+        case .initial:
+            return .secondary
+        }
+    }
+    
+    var label: String {
+        switch status {
+        case .initial:
+            return String(localized: "offline")
+        case .pending:
+            return String(localized: "syncing...")
+        case .failed:
+            return String(localized: "pending")
+        case .succeeded:
+            return String(localized: "up to date")
+        }
+    }
+    
+    @State var showLabel = true
+    
+    var body: some View {
+        HStack(spacing: AppTheme.unit2) {
+            Circle()
+                .fill(color)
+                .frame(width: 8, height: 8)
+        
+            Text(showLabel ? label : "")
```

Comment:
I was getting layout jitter if I actually remove the view rather than making it empty.

Reply:
That kind of scans. If the text stays, it has a stable ID in the underlying SwiftUI "DOM", and the tree is static. This usually means SwiftUI is able to be smarter about the view diffing behind the curtain.